### PR TITLE
Add xz/liblzma blog post

### PR DESCRIPTION
**Pull request by segiddins**

### CODE REVIEW

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+author: Samuel Giddins
+author_email: segiddins@segiddins.me
+---
+
+The past few days have seen the security world focused on the revelation of the [xy/liblzma backdoor](https://xeiaso.net/notes/2024/xz-vuln/), which has been assigned [CVE-2024-3094](https://nvd.nist.gov/vuln/detail/CVE-2024-3094).
+
+In response to this, I have undertaken an exposure assesment of the vulnerability on RubyGems.org and the broader RubyGems ecosystem.
+I am happy to report that RubyGems.org is not vulnerable to this issue, as we use a `musl` based operating system (Alpine) for our RubyGems.org containers.
+
+Furthermore, I am happy to confirm that there is no evidence that gems in the RubyGems ecosystem have been used an attack vector for this vulnerability.
```
Comment (martinemde) [inline _posts/2024-03-31-rubygems-and-xz.md:13]:
```suggestion
Furthermore, I am happy to confirm that there is no evidence that gems in the RubyGems ecosystem have been used as an attack vector for this vulnerability.
```

**Review by martinemde — APPROVED** [review]
Thanks for writing this up. Looks good with the small typo fix.

**Review by indirect — APPROVED** [review]
without any context or explanation, the phrase “as we use a musl based operating system (Alpine) for our RubyGems.org containers” comes across as a non-sequitur to me. Can we explain why that is relevant or worth mentioning?

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
+title: RubyGems & xz
+layout: post
+author: Samuel Giddins
+author_email: segiddins@segiddins.me
+---
+
+The past few days have seen the security world focused on the revelation of the [xy/liblzma backdoor](https://xeiaso.net/notes/2024/xz-vuln/), which has been assigned [CVE-2024-3094](https://nvd.nist.gov/vuln/detail/CVE-2024-3094).
```
Comment (duckinator) [inline _posts/2024-03-31-rubygems-and-xz.md:8]:
```suggestion
The past few days have seen the security world focused on the revelation of the [xz/liblzma backdoor](https://xeiaso.net/notes/2024/xz-vuln/), which has been assigned [CVE-2024-3094](https://nvd.nist.gov/vuln/detail/CVE-2024-3094).
```

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+In response to this, I have undertaken an exposure assesment of the vulnerability on RubyGems.org and the broader RubyGems ecosystem.
+I am happy to report that RubyGems.org is not vulnerable to this issue, as we use a `musl` based operating system (Alpine) for our RubyGems.org containers.
```
Comment (duckinator) [inline _posts/2024-03-31-rubygems-and-xz.md:11]: This feels a bit wordy -- maybe something closer to this?

```suggestion
I am happy to report that RubyGems.org is not vulnerable to this issue, as our servers use a `musl` based operating system (Alpine).
```

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Furthermore, I am happy to confirm that there is no evidence that gems in the RubyGems ecosystem have been used as an attack vector for this vulnerability.
+Thanks to the data consolidated in the [rubygems-research](https://github.com/segiddins/rubygems-research) project (available publicly at [research.rubygems.info](https://research.rubygems.info)), we can confirm that no gems contain any references to the vulnerable `liblzma` library.
+
+I would like to thank the rest of the RubyGems.org security team for their support in this investigation, and for their continued dedication to the security of the ecosystem. I would also like to thank AWS for their continued support of RubyGems security, sponsoring myself as [Ruby Central's security engineer in residence](https://rubycentral.org/news/ruby-central-welcomes-new-software-engineer-in-residence-sponsored-by-aws/), and funding the development of the [rubygems-research](https://github.com/segiddins/rubygems-research) project, which proved instrumental in confirming the absence of `liblzma` across all million and a half gem versions hosted on RubyGems.org.
+
+## Nitty Gritty Details
```
Comment (duckinator) [inline _posts/2024-03-31-rubygems-and-xz.md:18]: The tone feels off for the seriousness of the subject. Might make more sense to have it just be "Details".

```suggestion
## Details
```

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+RubyGems.org app containers do not contain any vulnerable versions of `liblzma` nor `xz` (in addition to the fact that they run Alpine and thus were not vulnerable to the backdoor).
```
Comment (duckinator) [inline _posts/2024-03-31-rubygems-and-xz.md:20]:
```suggestion
RubyGems.org app containers do not contain any vulnerable versions of `liblzma` nor `xz`. In addition, they run Alpine which is not vulnerable to the backdoor.
```

File: _posts/2024-03-31-rubygems-and-xz.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+```
+web-7b88594bd9-6g5nn:/app# find / -name '*xz*'
+/sys/module/xz_dec
+/usr/bin/unxz
+/usr/bin/xzcat
+/app/vendor/ruby/3.3.0/gems/bindata-2.5.0/lib/bindata/transform/xz.rb
+```
+
+No ecosystem gems contain a vulnerable `liblzma.so`.
```
Comment (duckinator) [inline _posts/2024-03-31-rubygems-and-xz.md:39]:
```suggestion
As of March 31st 2024, no gems on RubyGems.org contain a vulnerable `liblzma.so`.
```

**Review by duckinator — COMMENTED** [review]
I made some suggestions that I think make it a bit clearer. A lot of this is pretty subjective, though, so feel free to accept or reject individual changes as you like.

**Review by simi — APPROVED** [review]

**Pull request merged**